src/SparklineChart/SparklineChart.js

```diff
@@ -0,0 +1,457 @@
+import React, { createRef } from 'react';
+import PropTypes from 'prop-types';
+import * as d3 from 'd3';
+import { ChartTooltip } from './ChartTooltip';
+import { getDatasetMax, classSelector } from './utils';
+
+const LINE_WIDTH = 2;
+const AREA_MASK_ID = 'areaMaskId';
+const TOOLTIP_ELEMENT_RADIUS = 4;
+const DEFAULT_COLOR = '#3370FB';
+
+/** SparklineChart */
+class SparklineChart extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { getTooltipContent, highlightedStartingIndex } = props;
+
+    this.enableTooltip =
+      getTooltipContent && typeof getTooltipContent === 'function';
+    this.randomComponentId = Math.random().toString();
+    this.chartContext = {};
+
+    this.svgRef = createRef(null);
+    this.componentRef = createRef(null);
+    this.enableHighlightedAreaEffect = highlightedStartingIndex > 0;
+
+    this.state = {
+      hoveredLabels: [],
+    };
+  }
+
+  _useCreateContext = () => {
+    const halfWidth = LINE_WIDTH / 2;
+    const {
+      width = 200,
+      height = 40,
+      data,
+      highlightedStartingIndex = 0,
+      color = DEFAULT_COLOR,
+    } = this.props;
+
+    const margin = {
+      top: halfWidth + 2,
+      right: halfWidth,
+      bottom: halfWidth,
+      left: halfWidth,
+    };
+
+    const innerTop = margin.top;
+    const innerLeft = margin.left;
+    const innerHeight = height - innerTop - margin.bottom;
+    const innerWidth = width - innerLeft - margin.right;
+    const adaptedDataSet = {
+      values: this._getValues(data),
+    };
+    const minMax = getDatasetMax([adaptedDataSet]);
```

Comment:
Can be simply replaced by 

```suggestion
    const max = d3.max(adaptedDataSet, d => d.value);
    const min = d3.min(adaptedDataSet, d => d.value);
```

the utility can be removed as well.